pkg/pathutils/pathutils_test.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright © 2021 Aspect Build Systems Inc
+
+Not licensed for re-use.
+*/
+
+package pathutils
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	"github.com/spf13/cobra"
+)
+
+func getMockCmd(osGetwd func() (string, error)) *cobra.Command {
+	return &cobra.Command{
+		Use:   "mock",
+		Short: "This is a mock command",
+		Long:  "This is a mock command",
+		Args:  cobra.MaximumNArgs(1),
+		RunE: invokeCmdInsideWorkspace(
```

Comment:
Don't worry about trying to integrate with the user. From a unit test perspective, you don't care who your user is, you only care that the unitary logic behaves correctly with the input that you allow to be passed.

The tests should be split into 2 top-level `Test*` functions:
- TestWorkspaceFinder - you will mock the `osStat` to make it fail and assert that the `Find` method fails as expected, and play with the various possible `cwd` strings that could be passed to the `Find` method under ideal circumstances.
- TestInvokeInsideWorkspace - you will have to pass the `cobra.Command` to the returned function, which you will call it, but you will not assert that the `cmd` gets called. You will have to assert that the `cmd` gets passed to the `toRun`, though. You will never use the `defaultWorkspaceFinder` on your tests. You can create a `mock` directory and use the same pattern we've been using for other packages using [gomock](https://github.com/golang/mock).